cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractTwillProgramController.java

```diff
@@ -104,6 +111,32 @@ public void run() {
   }
 
   @Override
+  public void updateLogLevels(Map<String, LogEntry.Level> logLevels, @Nullable String componentName) throws Exception {
+    if (componentName == null) {
+      twillController.updateLogLevels(logLevels).get();
+    } else {
+      twillController.updateLogLevels(componentName, logLevels).get();
+    }
+  }
+
+  @Override
+  public void resetLogLevels(Set<String> loggerNames, @Nullable String componentName) throws Exception {
+    if (componentName == null) {
+      if (loggerNames.isEmpty()) {
+        twillController.resetLogLevels().get();
+      } else {
+        twillController.resetLogLevels(loggerNames.toArray(new String[0])).get();
```

Comment:
When creating an array for the `toArray` method, always new with the size of the collection.